app/models/package.rb

```diff
@@ -602,6 +608,30 @@ def donor_condition_name
 
   private
 
+  def create_and_save_copy
+    copy = dup
+    copy.quantity = 1
+    copy.received_quantity = 1
+    copy.inventory_number = generate_q_inventory_number
+    copy_and_save_images(copy) if copy.save
```

Comment:
How about calling `copy. copy_and_save_images    if copy.save` ?

Reply:
Then it would change the `self` object in `copy_and_save_images`. So I would have to pass `self` as an argument so either way it would turn out the same.

Reply:
Ohh Yes. 👍